File: original_scripts/r/binary_search/gemma_7b.r
```r
binary_search <- function(arr, target) {
  low <- 1
  high <- length(arr)
  while (low <= high) {
    mid <- floor((low + high) / 2)
    if (arr[mid] == target) {
      return(mid)
    } else if (arr[mid] < target) {
      low <- mid + 1
    } else {
      high <- mid - 1
    }
  }
  return(-1)
}
```

File: original_scripts/rust/binary_search/gemma_7b.rs
```rust
fn binary_search(arr: &[i32], target: i32) -> usize {
    let mut low = 0usize;
    let mut high = arr.len() - 1usize;

    while low <= high {
        let mid = (low + high) / 2usize;

        if arr[mid] == target {
            return mid;
        } else if arr[mid] < target {
            low = mid + 1usize;
        } else {
            high = mid - 1usize;
        }
    }

    -1usize
}
```